### ISSUE TITLE
[use-injectable-provided-in] Provide a way to forbid usage of providedIn

### ISSUE THREAD

**Issue by coyoteecd** [angular-eslint/angular-eslint#931]
**Description and reproduction of the issue**

One of the disadvantages of using `Injectable({ providedIn: 'root' })` is that it makes services available both to the Angular app, but also to the TestBed when executing unit tests. We have seen in our project multiple times that developers inject services in existing components and then forget to mock them. This leads to flaky, difficult-to-debug unit tests. A more extended discussion is available in the angular repo: [#25593](https://github.com/angular/angular/issues/25593).

Given that Angular's behavior is [unlikely to change](https://github.com/angular/angular/pull/26088), we want to enforce in our projects that nobody uses providedIn. The existing `use-injectable-provided-in` rule only checks that providedIn is specified. 

I see two options:
* adding a "mode" option that controls this behavior (with required/forbidden, or something similar, as values).
* adding a separate no-injectable-provided-in rule. One disadvantage I see is that it would be possible to have both rules in the same config file, which is a bit confusing (?).

Thoughts?

**Comment by JamesHenry** [comment]
Thanks @coyoteecd, I posted this same response on the Angular thread, but for completeness:

Creator of typescript-eslint and angular-eslint here 👋 

This was flagged up to me via a request to create a dedicated rule for this case. I am hesitant to do that for something which is still evolving as a topic and does not yet have a dedicated solution from the Angular Team.

However, you do not necessarily need to create a dedicated lint rule for this, the built-in ESLint `no-restricted-syntax` rule could be used to achieve the desired result:

```json
{
  "rules": {
    "no-restricted-syntax": [
      "error",
      {
        "selector": "Decorator[expression.callee.name=Injectable] Property[key.name=providedIn][value.value=root]",
        "message": "Do not use `providedIn: root` in Injectables"
      }
    ]
  }
}
```

Example of it in action:

https://typescript-eslint.io/play/#ts=4.5.2&sourceType=module&showAST=es&code=JYWwDg9gTgLgBAbzgSQHYCsCmBjGBDAIwBtM4BfOAMyghDgHIABPVAcwFci8oB6baTPQDcAKBGM0WXIRIAKBCLhK4YGgDdgAE0ya0ALgY0IMeiLIBKEdi4BnG3ABiECIjIigA&rules=N4Igdg9gtATgpgZwC4wJYGMlwCZQQTzCQEMAPEALgG0Q4YYIYQAaUBOAGzk0cpABFujYkkZU4pAA7wECVBDAA6dMQ5c4isMQC2cALwBJMACtuJAEZcAugAIbABQaS6SfFQDWcfJp37pEADdUbBwjKyoA1QBXDUiOGL0GCCQrFhBdWWIAczg+fggbSCQbKPYbAAN-IJDsIwobJKRym1QwGyNTTGJLRBAAXys+oA&tsConfig=N4IgpgHgDmBOCWBbMA7ALgQwDYBEwGMB7WDNYgZxAC41YBXMAGnEXjTyJLNgFkxMAJqQzVaDAL5A

![image](https://user-images.githubusercontent.com/900523/155580678-b72d4304-2398-4b55-a404-ac2b899568a5.png)
